cli/lsp/language_server.rs

```diff
@@ -19,38 +19,37 @@ use std::env;
 use std::path::PathBuf;
 use std::sync::Arc;
 use std::sync::Mutex;
+use std::sync::RwLock;
 use tokio::fs;
 
 use crate::deno_dir;
 use crate::import_map::ImportMap;
-use crate::media_type::MediaType;
 use crate::tsc_config::parse_config;
 use crate::tsc_config::TsConfig;
 
-use super::analysis;
 use super::capabilities;
 use super::config::Config;
 use super::diagnostics;
 use super::diagnostics::DiagnosticCollection;
 use super::diagnostics::DiagnosticSource;
-use super::memory_cache::MemoryCache;
+use super::documents::DocumentCache;
 use super::sources;
 use super::sources::Sources;
 use super::text;
-use super::text::apply_content_changes;
+use super::text::LineIndex;
 use super::tsc;
+use super::tsc::AssetDocument;
 use super::tsc::TsServer;
 use super::utils;
 
 #[derive(Debug, Clone)]
 pub struct LanguageServer {
-  assets: Arc<Mutex<HashMap<ModuleSpecifier, Option<String>>>>,
+  assets: Arc<RwLock<HashMap<ModuleSpecifier, Option<AssetDocument>>>>,
   client: Client,
   ts_server: TsServer,
   config: Arc<Mutex<Config>>,
-  doc_data: Arc<Mutex<HashMap<ModuleSpecifier, DocumentData>>>,
-  file_cache: Arc<Mutex<MemoryCache>>,
-  sources: Arc<Mutex<Sources>>,
+  documents: Arc<RwLock<DocumentCache>>,
+  sources: Arc<RwLock<Sources>>,
```

Comment:
I think it makes sense to use Mutex not RwLock.

Reply:
Yeah, I will revert that. Thanks.